academic_observatory_workflows/workflows/orcid_telescope.py

```diff
@@ -0,0 +1,756 @@
+
+        # Finish
+        self.add_task(self.add_new_dataset_release)
+        self.add_task(self.cleanup)
+
+        # The last task that the next DAG run's ExternalTaskSensor waits for.
+        self.add_operator(DummyOperator(task_id=external_task_id))
+
+    @property
+    def aws_orcid_key(self) -> Tuple[str, str]:
+        """Return API login and password"""
+        connection = BaseHook.get_connection(self.aws_orcid_conn_id)
+        return connection.login, connection.password
+
+    def make_release(self, **kwargs) -> OrcidRelease:
+        """Generates the OrcidRelease object."""
+        dag_run = kwargs["dag_run"]
+        is_first_run = is_first_dag_run(dag_run)
+        releases = get_dataset_releases(dag_id=self.dag_id, dataset_id=self.api_dataset_id)
+
+        # Determine the modication cutoff for the new release
+        if is_first_run:
+            if not len(releases) == 0 and kwargs["task"].task_id != self.cleanup.__name__:
+                raise ValueError(
+                    "fetch_releases: there should be no DatasetReleases stored in the Observatory API on the first DAG run."
+                )
+            prev_latest_modified_record = pendulum.instance(datetime.datetime.min)
+            prev_release_end = pendulum.instance(datetime.datetime.min)
+        else:
+            if not len(releases) >= 1:
+                raise ValueError(
+                    "fetch_releases: there should be at least 1 DatasetRelease in the Observatory API after the first DAG run"
+                )
+            prev_release = get_latest_dataset_release(releases, "changefile_end_date")
+            prev_release_end = prev_release.changefile_end_date
+            prev_latest_modified_record = pendulum.parse(prev_release.extra["latest_modified_record_date"])
+
+        return OrcidRelease(
+            dag_id=self.dag_id,
+            run_id=kwargs["run_id"],
+            cloud_workspace=self.cloud_workspace,
+            bq_dataset_id=self.bq_dataset_id,
+            bq_main_table_name=self.bq_main_table_name,
+            bq_upsert_table_name=self.bq_upsert_table_name,
+            bq_delete_table_name=self.bq_delete_table_name,
+            prev_release_end=prev_release_end,
+            start_date=kwargs["data_interval_start"],
+            end_date=kwargs["data_interval_end"],
+            prev_latest_modified_record=prev_latest_modified_record,
+            is_first_run=is_first_run,
+        )
+
+    def create_datasets(self, release: OrcidRelease, **kwargs) -> None:
+        """Create datasets"""
+
+        bq_create_dataset(
+            project_id=self.cloud_workspace.project_id,
+            dataset_id=self.bq_dataset_id,
+            location=self.cloud_workspace.data_location,
+            description=self.dataset_description,
+        )
+
+    def transfer_orcid(self, release: OrcidRelease, **kwargs):
+        """Sync files from AWS bucket to Google Cloud bucket."""
+        for i in range(self.transfer_attempts):
```

Comment:
`success` could be undefined if `self.transfer_attempts` was set to zero or negative, so better to set `success=False` before the loop.

Reply:
Yep, didn't think of this